sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/text/TextIOIT.java

```diff
@@ -127,28 +140,49 @@ public void writeThenReadAll() {
 
     PipelineResult result = pipeline.run();
     result.waitUntilFinish();
-    publishGcsResults(result);
+    gatherAndPublishMetrics(result);
   }
 
-  private void publishGcsResults(PipelineResult result) {
+  private void gatherAndPublishMetrics(PipelineResult result) {
+    String uuid = UUID.randomUUID().toString();
+    Timestamp timestamp = Timestamp.now();
+    List<NamedTestResult> namedTestResults = readMetrics(result, uuid, timestamp);
+    publishToBigQuery(namedTestResults, bigQueryDataset, bigQueryTable);
```

Comment:
We're missing a check here if the `bigQueryDataset` and `bigQueryTable` are present. Could you add one? 
We should not publish to bigQuery if user doesn't explicitly say so (by passing the options).

Reply:
Right, I missed it. Corrected